PyInstaller/building/utils.py

```diff
@@ -170,7 +170,15 @@ def checkCache(fnm, strip=False, upx=False, dist_nm=None):
         os.makedirs(cachedir)
     cacheindexfn = os.path.join(cachedir, "index.dat")
     if os.path.exists(cacheindexfn):
-        cache_index = load_py_data_struct(cacheindexfn)
+        try:
+            cache_index = load_py_data_struct(cacheindexfn)
+        except Exception as e:
+            # tell the user they may want to fix their cache
+            # .. however, don't delete it for them; if it keeps getting
+            #    corrupted, we'll never find out
+            logger.warn("pyinstaller bincache at `%s` may be corrupted; "
+                        "delete it and try again", cachedir)
```

Comment:
Please tell the user how to do this. (AFAIR `pyinstaller --clean` does the trick). But why not simply deleting it automatically, since it is of no use?